### DISCUSSION TITLE
How do you use electron-log ?

### DISCUSSION THREAD

**Discussion by ajoslin103** [megahertz/electron-log#306] [Q&A]
---  Ideally this discussion would serve as examples and samples for people who would chuck wood [at logs]

I use it thusly - so far

src/global.d.ts
```
import Logger from "electron-log";
declare global {
  var mainLogger: Logger.ElectronLog;
}
export {};
```

src/common/logger-main.ts
```
const Logger = require("electron-log")
const loggerMain = Logger.create('main')
loggerMain.transports.file.fileName = "main.log";
loggerMain.transports.file.maxSize = 1024 * 1024;
loggerMain.level = 'debug';
global.mainLogger = loggerMain;
```

src/main.ts
```
import "../common/logger-main"
```

src/appWindows.ts
```
const logger = global.mainLogger;
[...]
logger.debug(`createNewWindow newWindow.on("closed`);
```

**Comment by vladpereverzyev** [comment]
Your approach can work, especially if you want a dedicated logger for the Electron main process.

A couple of suggestions, though:

1. You don't necessarily need to expose the logger through `global`. If the logger is imported from a shared module, you can export the configured instance instead:

```ts
// logger-main.ts
import log from "electron-log";

log.transports.file.fileName = "main.log";
log.transports.file.maxSize = 1024 * 1024;
log.level = "debug";

export default log;
```

Then:

```ts
// appWindows.ts
import logger from "./common/logger-main";

logger.debug('Creating new window');
```

This makes the dependency explicit and avoids relying on a global variable.

2. If you do want separate loggers for different parts of the application, `Logger.create("main")` is a reasonable approach. You can configure the transports and levels on that instance and then share it with the relevant modules.

3. For TypeScript, you should also be careful with the type declaration. Since the logger is already exported by `electron-log`, using an exported/shared instance generally avoids having to augment `global` just for the logger.

So the example in the question is valid, but I would generally prefer exporting the configured logger from a module rather than storing it on `global`. It makes the code easier to test and makes it clear where the logger comes from.
